### PR TITLE
Allow binary sensor device class updates

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Allows choosing between all device classes in the binary sensor settings

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -786,7 +786,31 @@
               "door": "Door",
               "garage_door": "Garage door",
               "window": "Window",
-              "opening": "Other"
+              "opening": "Other",
```
Comment (balloob) [inline src/translations/en.json:789]: Opening = Other ?

File: src/panels/config/entities/entity-registry-settings.ts [hunk 2]
```diff
@@ -62,7 +62,36 @@ const OVERRIDE_DEVICE_CLASSES = {
     "shutter",
     "window",
   ],
-  binary_sensor: ["window", "door", "garage_door", "opening"],
+  binary_sensor: [
```
Comment (balloob) [inline src/panels/config/entities/entity-registry-settings.ts:65]: Can we only show every possible option when you're in advanced mode?

File: src/translations/en.json (hunk 1 above)
Comment (zsarnett) [reply src/translations/en.json:789]: 🤷‍♂️  It was this way before haha. I guess opening is an interesting device class

File: src/panels/config/entities/entity-registry-settings.ts (hunk 2 above)
Comment (zsarnett) [reply src/panels/config/entities/entity-registry-settings.ts:65]: Yep. Need to set that up. Also need to figure out the categories

File: src/panels/config/entities/entity-registry-settings.ts [hunk 3]
```diff
@@ -42,27 +48,39 @@ import { SubscribeMixin } from "../../../mixins/subscribe-mixin";
...
+      "door",
+      "garage",
+      "gate",
+      "shade",
+      "shutter",
+      "window",
+    ],
+  ],
+  binary_sensor: [
+    ["connectivity", "lock", "update"], // Other
```
Comment (balloob) [inline src/panels/config/entities/entity-registry-settings.ts:68]: `connectivity` and `update`

I don't see how this ever can be miscategorized and the user needs to edit it… do we really need to cover ALL binary sensor device classes ?

File: src/panels/config/entities/entity-registry-settings.ts [hunk 4]
```diff
@@ -127,21 +147,39 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
...
-      this._hiddenBy = this.entry.hidden_by;
-      this._device =
-        this.entry.device_id && this._deviceLookup
-          ? this._deviceLookup[this.entry.device_id]
-          : undefined;
+    if (!changedProperties.has("entry")) {
+      return;
+    }
+
+    this._error = undefined;
```
Comment (balloob) [inline src/panels/config/entities/entity-registry-settings.ts:154]: All these things will cause a new re-render. That means the changes need to be made in `willUpdate`.

**Review by balloob — DISMISSED** [review]

File: src/panels/config/entities/entity-registry-settings.ts (hunk 3 above)
Comment (zsarnett) [reply src/panels/config/entities/entity-registry-settings.ts:68]: Update I can see

But I could see Connectivity being needed?

Comment (balloob) [reply src/panels/config/entities/entity-registry-settings.ts:68]: I guess if you want to change connectivity to be problem?

Comment (zsarnett) [reply src/panels/config/entities/entity-registry-settings.ts]: I took it out. I guess lets see if it comes up

**Review by balloob — APPROVED** [review]

**Pull request merged**